src/apps/origin-dollar/ethereum/origin-dollar.veogv.token-fetcher.ts

```diff
@@ -0,0 +1,129 @@
+ * @param totalSupplyVeOGV {BigNumber} - The total supply of veOGV
+ * @return {number} A decimal representation of expected APY for a 4 year stake
+ */
+function getRewardsApy(ogv, veogv, totalSupplyVeOGV) {
+  const pctShare = veogv.mul(oneEther).div(totalSupplyVeOGV.add(veogv));
+  const now = stamp();
+  const dailyEmissionsRow = dailyEmissionsTable.find(([startTime, endTime]) => now > startTime && now < endTime);
+  const dailyEmissions = dailyEmissionsRow ? dailyEmissionsRow[2] : 0;
+  const dailyRewards = toBN(dailyEmissions).mul(pctShare);
+  const yearlyRewards = dailyRewards.mul(floatToBN(365.25)).div(oneEther);
+  const apr = yearlyRewards.mul(oneEther).div(ogv);
+  return parseFloat(format(apr));
+}
+
+@Register.TokenPositionFetcher({ appId, groupId, network })
+export class EthereumOriginDollarVeogvTokenFetcher implements PositionFetcher<AppTokenPosition> {
+  constructor(
+    @Inject(APP_TOOLKIT) private readonly appToolkit: IAppToolkit,
+    @Inject(OriginDollarContractFactory) private readonly originDollarContractFactory: OriginDollarContractFactory,
+  ) {}
+
+  async getPositions() {
+    const baseTokens = await this.appToolkit.getBaseTokenPrices(network);
+    const multicall = this.appToolkit.getMulticall(network);
+
+    const ogv = baseTokens.find(v => v.address === '0x9c354503c38481a7a7a51629142963f98ecc12d0');
+    if (!ogv) return [];
+
+    const contract = this.originDollarContractFactory.veogv({
+      address: '0x0c4576ca1c365868e162554af8e385dc3e7c66d9',
+      network,
+    });
+
+    // Amount we'll use to calculate APY (1000 OGV)
+    const stakeAmount = ethers.BigNumber.from('1000000000000000000000');
+
+    // Get contract data
+    const [supplyRaw, [expectedVeOGV], ogvBalance] = await Promise.all([
+      multicall.wrap(contract).totalSupply(),
+      // 1000e18 @ 4 year stake (1461 days)
+      multicall.wrap(contract).previewPoints(stakeAmount, ethers.BigNumber.from('126230400')),
+      multicall
+        .wrap(this.appToolkit.globalContracts.erc20({ network, address: ogv.address }))
+        .balanceOf(contract.address),
+    ]);
+
+    // Calculate TVL in USD
+    const supply = parseFloat(format(supplyRaw));
+    const totalValueLocked = parseFloat(
+      format(supplyRaw.mul(ethers.utils.parseUnits(ogv.price.toString(), 18)).div(oneEther)),
+    );
+    const apy = getRewardsApy(stakeAmount, expectedVeOGV, supplyRaw);
+
+    // Approximate value of position
+    const ratio = parseFloat(format(supplyRaw.mul(oneEther).div(ogvBalance)));
+    const price = ogv.price / ratio;
+
+    const token: AppTokenPosition<VeOGVTokenDataProps> = {
+      type: ContractType.APP_TOKEN,
+      appId,
+      groupId,
+      address: contract.address,
+      network,
+      symbol: 'veOGV',
+      decimals: 18,
+      supply,
+      tokens: [ogv],
+      pricePerShare: [price],
+      price,
+      dataProps: {
+        apy,
+        totalValueLocked,
```

Comment:
Rename as `liquidity` so that our system can pick it up in the DeFi explore section in Zapper.